python/cudf/cudf/core/series.py

```diff
@@ -6156,8 +6166,18 @@ def is_quarter_end(self):
         7    False
         dtype: bool
         """
-        return (self.is_month_end & self.month.isin([3, 6, 9, 12])).fillna(
-            False
+        day = self.series._column.get_dt_field("day")
+        last_day = libcudf.datetime.last_day_of_month(self.series._column)
+        last_day = last_day.get_dt_field("day")
+        last_month = self.series._column.get_dt_field("month").isin(
+            [3, 6, 9, 12]
+        )
+
+        result = ((day == last_day) & last_month).fillna(False)
+        return Series._from_data(
+            ColumnAccessor({None: result}),
```

Comment:
We can just pass `dict` object to `_from_data` without constructing `ColumnAccessor`

Reply:
Fixed